controllers/nodemaintenance_controller.go

```diff
@@ -135,13 +139,15 @@ func (r *NodeMaintenanceReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		// Stop node maintenance - uncordon and remove live migration taint from the node.
 		if err := r.stopNodeMaintenanceOnDeletion(ctx, drainer, nm.Spec.NodeName); err != nil {
 			r.logger.Error(err, "error stopping node maintenance")
-			if !errors.IsNotFound(err) {
+			if !apiErrors.IsNotFound(err) {
 				return r.onReconcileError(ctx, nm, drainer, err)
 			}
 		}
 
 		// Remove finalizer
 		controllerutil.RemoveFinalizer(nm, v1beta1.NodeMaintenanceFinalizer)
+		// end maintenance on removing finalizer, taints, and node is already uncordoned
+		utils.NormalEvent(r.Recorder, nm, utils.EventReasonRemovedMaintenance, utils.EventMessageRemovedMaintenance)
```

Comment:
Same comment as above, IMO better to omit the event after the action is successful.